File: app/src/main/java/com/prography/prography_androidstudy/Room/UserModel.java
package com.prography.prography_androidstudy.Room;

import android.content.Context;

import java.util.ArrayList;

public class UserModel {
    private UserDatabase db;
    private UserDao userDao;

    public UserModel(Context context) {
        db = UserDatabase.getDatabase(context);
        userDao = db.userDao();
    }

    public boolean checkLogin(final String email, final String pw) {
        final ArrayList<User> userArrayList = new ArrayList<>();
        Thread loginThread = new Thread(new Runnable() {
            @Override
            public void run() {
                userArrayList.addAll(userDao.userLogin(email, pw));
            }
        });
        loginThread.start();

        try {
            loginThread.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        if (userArrayList.size() == 0) {
            return false;
        } else {
            return userArrayList.get(0).getEmail().equals(email);
        }
    }

    public String signUp(final String email, String pw, String pwCheck) {
        if (!pw.equals(pwCheck)) {
            return "PasswordError";
        }

        final ArrayList<User> userArrayList = new ArrayList<>();
        Thread signUpThread = new Thread(new Runnable() {
            @Override
            public void run() {
                userArrayList.addAll(userDao.findUser(email));
            }
        });
        signUpThread.start();

        try {
            signUpThread.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        /* User Data 중복 */
        if (userArrayList.size() != 0) {
            return "RedundantUser";
        }

        final User newUser = new User(email, pw);
        Thread uploadUserThread = new Thread(new Runnable() {
            @Override
            public void run() {
                userDao.insert(newUser);
            }
        });
        uploadUserThread.start();
        return "Success";
    }
}

File: app/src/main/java/com/prography/prography_androidstudy/Model/UserModel.kt
package com.prography.prography_androidstudy.Model

import android.content.Context
import com.prography.prography_androidstudy.Room.User
import com.prography.prography_androidstudy.Room.UserDao
import com.prography.prography_androidstudy.Room.UserDatabase

class UserModel(context: Context) {
    private var database: UserDatabase = UserDatabase.getInstance(context)
    private var userDao: UserDao

    init {
        userDao = database.userDao
    }

    fun checkLogin(email: String, pw: String): Boolean {
        val userList = ArrayList<User>()
        val loginThread = Thread {
            userList.addAll(userDao.userLogin(email, pw))
        }
        loginThread.start();

        try {
            loginThread.join()
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        return if (userList.size == 0) {
            false
        } else userList[0].email == email
    }


    fun signUp(email: String, pw: String, pwCheck: String): String {
        val userList = ArrayList<User>()
        val signUpThread = Thread {
            userList.addAll(userDao.findUser(email))
        }
        signUpThread.start()

        try {

        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        if (userList.size != 0) {
            return "Already"
        }

        if (pw != pwCheck) {
            return "NotChecked"
        }

        val user = User(email, pw)
        Thread {
            database.userDao.insert(user)
        }.start()

        return "Success"
    }

}